### PR TITLE
[backup] metadata cache: download to tmp file then rename

### PR DESCRIPTION
**Pull request by msmouse**
## Motivation

so that if error happens we don't end up with a stale (probably empty) file cached.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/diem/diem/blob/master/CONTRIBUTING.md#pull-requests)?

Y

## Test Plan

locally

### CODE REVIEW

File: storage/backup/backup-cli/src/metadata/cache.rs [hunk 1]
```diff
@@ -110,6 +110,8 @@ pub async fn sync_and_load(
         async move {
             let file_handle = fh_by_h_ref.get(*h).expect("In map.");
             let local_file = cache_dir_ref.join(*h);
+            let local_tmp_file = cache_dir_ref.join(format!(".{}", *h));
```
Comment (wqfish) [inline storage/backup/backup-cli/src/metadata/cache.rs:113]: this is adding a `.`?

File: storage/backup/backup-cli/src/metadata/cache.rs [hunk 2]
```diff
@@ -118,11 +120,14 @@ pub async fn sync_and_load(
                 &mut OpenOptions::new()
                     .write(true)
                     .create_new(true)
-                    .open(&local_file)
+                    .open(&local_tmp_file)
                     .await
                     .err_notes(&local_file)?,
             )
             .await?;
+            // rename to target file only if successful; stale tmp file caused by failure will be
```
Comment (wqfish) [inline storage/backup/backup-cli/src/metadata/cache.rs:128]: could also use tmpfs?

**Review by wqfish — APPROVED** [review]

File: storage/backup/backup-cli/src/metadata/cache.rs (hunk 1 above)
Comment (msmouse) [reply storage/backup/backup-cli/src/metadata/cache.rs:113]: Yes

File: storage/backup/backup-cli/src/metadata/cache.rs (hunk 2 above)
Comment (msmouse) [reply storage/backup/backup-cli/src/metadata/cache.rs:128]: Transferring the file from tmpfs to the target folder is slower and tends more to fail than a renaming?

**Pull request merged**